packages/common/src/TransactionUtils.ts

```diff
@@ -30,26 +43,30 @@ const argv = minimist(process.argv.slice(), {});
  * @param {*Object} transaction Transaction to call `.call()` and subsequently `.send()` on from `senderAccount`.
  * @param {*Object} transactionConfig config, e.g. { maxFeePerGas, maxPriorityFeePerGas, from } or { gasPrice, from}
  *     depending if this is a london or pre-london transaction, passed to transaction.
+ * @param availableAccounts {number} defines how many EOAs the transaction runner has access too. If the 0th account
+ * has a pending tx then the runner will automatically send the transaction from the next EOA.
+ * @param waitForMine {Boolean} informs if the transaction runner should wait until the tx is mined or return early once
+ * it has a transaction hash. Useful when sending many transactions in quick succession.
  * @return Error and type of error (originating from `.call()` or `.send()`) or transaction receipt and return value.
  */
 export const runTransaction = async ({
   web3,
   transaction,
   transactionConfig,
   availableAccounts = 1,
+  waitForMine = true,
 }: {
   web3: Web3;
   transaction: ContractSendMethod;
   transactionConfig: AugmentedSendOptions;
   availableAccounts?: number;
-}): Promise<{
-  receipt: TransactionReceipt;
-  returnValue: CallReturnValue;
-  transactionConfig: AugmentedSendOptions;
-}> => {
+  waitForMine?: boolean;
+}): Promise<executedTransaction> => {
+  // use a cast version of web3 to enable callers to not have to define the `nonces` mapping in the web3 object.
+  const augmentedWeb3 = web3 as AugmentedWeb3;
```

Comment:
@mrice32 lmk if this is what you meant. problem is we need the interface on this method to take in a normal web3 object which we then append the `nonces` onto.

Reply:
Yes, this is exactly what I meant. To keep the web3 name, you might be able to remap the name of the web3 object that the method takes in like this:
```
export const runTransaction = async ({
  web3: _web3,
  transaction,
  ...

  const web3 = web3_ as AugmentedWeb3;
```

Just to reduce confusion and keep the usage consistent. Up to you, though.